pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -504,10 +506,6 @@ func completeLimit(existingLimit *v32.ContainerResourceLimit, defaultLimit *v32.
 		}
 	}
 
-	if reflect.DeepEqual(existingLimitMap, newLimitMap) {
-		return nil, nil
-	}
-
```

Comment:
I'm not exactly sure why you removed this, would you mind elaborating?

Near as I can tell, this caused the code to report a nil 'combined map' in cases where the namespace limit and the combined map (produced by replacing all values in the project limit whose keys are also in the namespace limit) are identical. This would cause the calling function (`getResourceLimitToUpdate`) to use the namespace limit. This new case functions identically, except that we return up the combined map from this function (which is the same as the namespace limit). Was this more of an efficiency change?

Reply:
This is for the following scenario: there are project's container limits and a namespace's overridden values. This unnecessary map equality check prevents the override from being saved (because nil is returned). Removing this makes sure a namespace's overrides are actually honored.

I tested different scenarios:
- project container limit **defined**, namespaces override **defined**
- project container limit **defined**, namespaces override **undefined**
- project container limit **undefined**, namespaces override **defined**
- project container limit **undefined**, namespaces override **undefined**

Reply:
Here's my confusion - in the case that you described, this function would return nil (since the namespace's values were all that ended up being used). Since there are no combined values, the calling function would then use the values from the namespace. Which is the same as what will be returned from the new function. Correct me if I am wrong there. Would you mind explaining what you see as old/new side for that scenario that you described?

Reply:
You're right, I just saw a flaw in my tests. Re-tested again with the map equality comparison. Will update the code.